src/Microsoft.AspNetCore.Authentication.Impl/AuthenticationServiceCollectionExtensions.cs

```diff
@@ -46,8 +45,7 @@ public static IServiceCollection AddAuthentication(this IServiceCollection servi
             return services;
         }
 
-        // REVIEW: rename to just AddScheme?
-        public static IServiceCollection AddSchemeHandler<TOptions, THandler>(this IServiceCollection services, string authenticationScheme, Action<TOptions> configureOptions)
+        public static IServiceCollection AddScheme<TOptions, THandler>(this IServiceCollection services, string authenticationScheme, Action<TOptions> configureOptions, bool canHandleRequests)
```

Comment:
@HaoK point taken, tho' I'm not sure why you'd want to support scenarios with "hundreds/thousands of schemes" if there's no real support in the options stack (I'm not sure having hundreds of schemes + hundreds of handlers + hundreds of options instances is really _scalable_).

Not being able to invoke my handler for every path is really annoying for me. As a compromise, would you consider bringing back your per-scheme "promiscuous mode"? (aka `CanHandleRequest` :smile:)